react/AppIcon/Preloader.js

```diff
@@ -46,6 +46,10 @@ const _getInstalledIconPath = app => {
 }
 
 const _getRegistryIconPath = app => {
+  if (typeof app === 'string') {
```

Comment:
I am not a huge fan of having a prop which can have several types but it seems the better thing to do here to avoid a huge refactoring.

Reply:
Yep. I think it's better to have a single prop that can be of multiple types than to have several props that would make impossible states possibles (or more complicated to validate).

With a single prop, we just have to check its type. With multiple props, we may end with `app={{ slug: 'stuff', latest_version: '1.2.3' }}` and `slug="thing"` and the same time. In that case we have to decide which one to take. It can be complicated IMHO.